tikv/cop_handler_test.go

```diff
@@ -0,0 +1,363 @@
+	// the reason that we don't make a tableScanExecutor directly here is
+	// it would be replaced by closureExecutor in the future, so, for compatibility,
+	// we just build a dagReq, and call buildDAGExecutor to build
+	// the corresponding executor.
+	aRange := getTestPointRange(TableId, handler)
+	dagReq := &tipb.DAGRequest{
+		StartTs: uint64(time.Now().UnixNano()),
+		Executors: [] *tipb.Executor{
+			{
+				Tp: tipb.ExecType_TypeTableScan,
+				TblScan: &tipb.TableScan{
+					Columns: data.colInfos,
+					TableId: TableId,
+				},
+			},
+		},
+		// only want first two cols.
+		OutputOffsets: [] uint32{
+			uint32(0),
+			uint32(1),
+		},
+	}
+	sc := flagsToStatementContext(dagReq.Flags)
+	sc.TimeZone = time.FixedZone("UTC", int(dagReq.TimeZoneOffset))
+	dagCtx := &dagContext{
+		reqCtx: &requestCtx{
+			svr:   store.svr,
+			dbIdx: TableId,
+			regCtx: &regionCtx{
+				meta: &metapb.Region{
+					StartKey: nil,
+					EndKey:   nil,
+				},
+			},
+		},
+		dagReq: dagReq,
+		keyRanges: []*coprocessor.KeyRange{
+			{
+				Start: aRange.StartKey,
+				End:   aRange.EndKey,
+			},
+		},
+		evalCtx: &evalContext{sc: sc},
+	}
+	dagCtx.evalCtx.setColumnInfo(dagReq.Executors[0].TblScan.Columns)
+	closureExec, error := store.svr.tryBuildClosureExecutor(dagCtx, dagReq)
+	if error != nil {
+		return nil, error
+	}
+	var rowCnt int
+	if closureExec != nil {
+		chunks, error = closureExec.execute()
+		if error != nil {
+			return nil, error
+		}
+		return chunks, nil
+	}
+	e, error := store.svr.buildDAGExecutor(dagCtx, dagReq.Executors)
+	if error != nil {
+		return nil, error
+	}
+	ctx := context.TODO()
+	for {
+		var row [][]byte
+		e.Counts()
+		row, err = e.Next(ctx)
+		if err != nil {
+			break
+		}
+		if row == nil {
+			break
+		}
+		data := dummySlice
+		for _, offset := range dagReq.OutputOffsets {
+			data = append(data, row[offset]...)
+		}
+		chunks = appendRow(chunks, data, rowCnt)
+		rowCnt++
+	}
+	return chunks, nil
+}
+
+// see tikv/src/coprocessor/util.rs for more detail
+func TestIsPrefixNext(t *testing.T) {
+	require.True(t, isPrefixNext([]byte{}, []byte{0}))
+	require.True(t, isPrefixNext([]byte{0}, []byte{1}))
+	require.True(t, isPrefixNext([]byte{1}, []byte{2}))
+	require.True(t, isPrefixNext([]byte{255}, []byte{255, 0}))
+	require.True(t, isPrefixNext([]byte{255, 255, 255}, []byte{255, 255, 255, 0}))
+	require.True(t, isPrefixNext([]byte{1, 255}, []byte{2, 0}))
+	require.True(t, isPrefixNext([]byte{0, 1, 255}, []byte{0, 2, 0}))
+	require.True(t, isPrefixNext([]byte{0, 1, 255, 5}, []byte{0, 1, 255, 6}))
+	require.True(t, isPrefixNext([]byte{0, 1, 5, 255}, []byte{0, 1, 6, 0}))
+	require.True(t, isPrefixNext([]byte{0, 1, 255, 255}, []byte{0, 2, 0, 0}))
+	require.True(t, isPrefixNext([]byte{0, 255, 255, 255}, []byte{1, 0, 0, 0}))
+}
+
+func TestPointGet(t *testing.T) {
+	var (
+		data   *data
+		store  *TestStore
+		err    error
+		errors []error
+		chunks []tipb.Chunk
+		row    []types.Datum
+		l      int
```

Comment:
Better define the local variable where it's first used.
And what does 'l' mean?

Reply:
oops, my fault, I would change the name of this parameter and define it as a local variable